shell/server/core.js

```diff
@@ -147,6 +148,46 @@ class NotificationHandle {
   }
 }
 
+class PersistentSealedUiViewImpl {
+  constructor(token, membraneRequirements) {
+    this._token = token;
+    // TODO: implement SystemPersistent.addRequirements() in a way that collects requirements, and
+    // then saves them appropriately in save().
+    this._membraneRequirements = membraneRequirements || [];
+  }
+
+  save(params) {
+    // Create a new child apiToken for the user whose session attempted to save this capability.
+    // The token should specify this.token as the parent token, and needs to gain an additional membrane
+    // requirement (TODO):
+    // * if params.sealFor is a user, then the user must still have access to the grain that offered
+    //   this cap.  Today, this is handled by offer() in SessionContextImpl.
+    const res = inMeteor(() => {
+      const owner = _.extend({}, params.sealFor);
+      const grainId = this._token.grainId;
+
+      if (owner.user) {
+        // Copy the owner's save label for this capability, if available, to be the new title.
+        if (this._token && this._token.owner && this._token.owner.grain &&
+            this._token.owner.grain.saveLabel && this._token.owner.grain.saveLabel.defaultText) {
+          owner.user.title = this._token.owner.grain.saveLabel.defaultText;
+        }
+
+        // Denormalize this grain's app metadata into owner.user.
+        owner.user.denormalizedGrainMetadata = globalDb.getDenormalizedGrainInfo(grainId);
+      }
+
+      const ret = makeChildTokenInternal(this._token._id, owner, this._membraneRequirements, grainId);
```

Comment:
See [the documentation for the `parentToken` field](https://github.com/sandstorm-io/sandstorm/blob/618193034f4fcd5d45e0f720b783ab61a68878f7/shell/packages/sandstorm-db/db.js#L365-L37) --- in addition to the `grainId` field, we should make sure that the child inherits the `identityId` and `accountId` fields of the parent.

Maybe the first argument to `makeChildTokenInternal` should be the actual entry in `ApiTokens` representing the parent?